pkg/syncer/metadatasyncer.go

```diff
@@ -831,6 +842,345 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
+	volumeMap map[string]*v1.PersistentVolume, namespace string, metadataSyncer *metadataSyncInformer) (map[string]*resource.Quantity, error) {
+	log := logger.GetLogger(ctx)
+	// get storageclass and storage policy ID mappings.
+	scToStoragePolicyIDMap, err := fetchStorageClassToStoragePolicyMapping(ctx, cnsOperatorClient)
+	if err != nil {
+		return nil, err
+	}
+	// fetch pvc for given namespace
+	log.Infof("Fetching PersistentVolumeClaim for namespace %q", namespace)
+	pvcList, err := metadataSyncer.pvcLister.PersistentVolumeClaims(namespace).List(nil)
+	if err != nil {
+		log.Errorf("unable to fetch all pvc from namespace %q, Error %v", namespace, err)
+		return nil, err
+	}
+	storagePolicyToReservedMap := make(map[string]*resource.Quantity)
+	for _, pvc := range pvcList {
+		log.Infof("Processing PersistentVolumeClaim Name: %q, Namespace: %q", pvc.Name, pvc.Namespace)
+		if pvc.DeletionTimestamp != nil {
+			log.Infof("pvc is marked for deletion, ignoring Name: %q, Namespace: %q",
+				pvc.Name, pvc.Namespace)
+			continue
+		}
+		if pvc.Status.Phase == "" {
+			log.Infof("pvc status not populated, continuing processing others Name: %q, Namespace: %q",
+				pvc.Name, pvc.Namespace)
+			continue
+		}
+		if storagePolicyID, ok := scToStoragePolicyIDMap[*pvc.Spec.StorageClassName]; ok {
+			if storagePolicyToReservedMap[storagePolicyID] == nil {
+				storagePolicyToReservedMap[storagePolicyID] = resource.NewQuantity(0, resource.BinarySI)
+			}
+		}
+		// check if pvc is in pending state
+		pvcSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvc.Status.Phase == v1.ClaimPending {
+			if storagePolicyID, ok := scToStoragePolicyIDMap[*pvc.Spec.StorageClassName]; ok {
+				log.Infof("pvc is in pending state, Name: %q, Namespace: %q adding pvc capacity to reserved",
+					pvc.Name, pvc.Namespace)
+				storagePolicyToReservedMap[storagePolicyID].Add(pvcSize)
+			}
+		} else if pv, ok := volumeMap[pvc.Name]; ok {
+			pvSize := pv.Spec.Capacity[v1.ResourceStorage]
+			// check if pvc is under expansion
+			if !pvcSize.Equal(pvSize) {
+				log.Infof("pvc size being expanded Name: %q, Namespace: %q adding pvc capacity to reserved",
+					pvc.Name, pvc.Namespace)
+				if storagePolicyID, ok := scToStoragePolicyIDMap[*pvc.Spec.StorageClassName]; ok {
+					// get accurate expected reserved value
+					pvcSize.Sub(pvSize)
+					storagePolicyToReservedMap[storagePolicyID].Add(pvcSize)
+				}
+			}
+		}
+	}
+	return storagePolicyToReservedMap, nil
+}
+
+// calculateVolumeSnapshotReservedForNamespace calculate the expected reserved value for volumesnapshot for namespace
+func calculateVolumeSnapshotReservedForNamespace(ctx context.Context, cnsOperatorClient client.Client,
+	namespace string, metadataSyncer *metadataSyncInformer) (map[string]*resource.Quantity, error) {
+	log := logger.GetLogger(ctx)
+	scTospuMap, err := fetchStorageClassToStoragePolicyMapping(ctx, cnsOperatorClient)
```

Comment:
Make changes to map names scTospuMap and spuWiseReservedMap in this function as well